spec/models/spree_avatax_certified/soap_api_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+
+describe SpreeAvataxCertified::SoapApi, type: :model do
+  let!(:soap) { SpreeAvataxCertified::SoapApi.new }
+
+  describe '#ping' do
+    it 'responds with a hash' do
+      expect(soap.ping).to be_kind_of(Hash)
+    end
+    it 'responds successfully' do
+      expect(soap.ping[:ping_response][:ping_result][:result_code]).to eq('Success')
+    end
+  end
+
+  describe '#adjust_tax' do
+    let(:taxsvc) { TaxSvc.new }
+    code = rand(100000).to_s
+    let(:request_hash) { attributes_for(:request_hash, Commit: true, DocType: 'SalesInvoice', DocCode: code, ReferenceCode: code) }
+
+    before do
+      taxsvc.get_tax(request_hash)
+    end
+
+    subject { soap.adjust_tax(request_hash) }
+    it 'should return a hash' do
+      expect(subject).to be_kind_of(Hash)
+    end
+
+    it 'should have result code equal success' do
+      expect(subject[:adjust_tax_response][:adjust_tax_result][:result_code]).to eq('Success')
+    end
+  end
+end
```

Comment:
Not sure about the particular case, but usually with 3rd party integrations, the `vcr` gem  gets to be quite useful. There's a few examples of it in Magpie.